src/main/dialogs/promotionWindowDialog.js

```diff
@@ -0,0 +1,192 @@
+                    }
+                }
+            }
+        },
+        repositionWindowTimeout: {
+            type: "gpii.app.timer",
+            options: {
+                listeners: {
+                    onTimerFinished: {
+                        funcName: "{promotionWindowDialog}.showPromotionWindow",
+                        args: [true]
+                    }
+                }
+            }
+        }
+    },
+    invokers: {
+        showPromotionWindow: {
+            funcName: "gpii.app.promotionWindowDialog.show",
+            args: [
+                "{promotionWindowDialog}",
+                "{showTimer}",
+                "{promotionWindowDialog}.options.siteConfig.offset.x",
+                "{promotionWindowDialog}.options.siteConfig.offset.y",
+                "{tray}",
+                "{arguments}.0"
+            ]
+        }
+    }
+});
+
+
+/**
+ * Shows the promotion window dialog. The '{promotionWindowDialog}.show' function
+ * is invoked through the 'showTimer' component. If a delay is not specified
+ * the promotion dialog will be shown without delay. This function also set the position
+ * of the promotion dialog with given 'x' and 'y' offset. If offset 'x' and 'y' are not given
+ * the window will be displayed with the default position (0, 0). If the 'centered' option
+ * is specified, the dialog will be positioned in the center of the screen.
+ * The same function will be invoked when repositioning of the promotion window dialog is required.
+ * @param {gpii.app.promotionWindowDialog} that - The instance of the widget.
+ * @param {gpii.app.timer} timer - An instance of `gpii.app.timer` used for showing
+ * the widget with a delay.
+ * @param {Integer} offsetX - The x offset from the right edge of the screen.
+ * @param {Integer} offsetY - The y offset from the bottom edge of the screen.
+ * @param {gpii.app.tray} tray - The `gpii.app.tray` instance.
+ * @param {Boolean} reposition - Whether need only reposition of the window.
+ */
+gpii.app.promotionWindowDialog.show = function (that, timer, offsetX, offsetY, tray, reposition) {
+    if (that.options.siteConfig.positionByTrayIcon) {
+        // getting the tray position only when it's required (siteConfig > positionByTrayIcon)
+        var trayPosition = tray.getIconBounds(),
+            screen = require("electron").screen,
+            taskBarHeight = screen.getPrimaryDisplay().bounds.height - screen.getPrimaryDisplay().workAreaSize.height,
+            scaleFactor = screen.getPrimaryDisplay().scaleFactor,
+            displaySize = { width: screen.getPrimaryDisplay().workAreaSize.width, height: screen.getPrimaryDisplay().workAreaSize.height - taskBarHeight},
+            offset = { x: displaySize.width - ((trayPosition.x + (trayPosition.width / 2)) / scaleFactor), y: displaySize.height - ((trayPosition.y - (trayPosition.height / 2)) / scaleFactor) };
+
+        // setting the position next to the tray icon
+        that.setPosition(offset.x, offset.y);
```

Comment:
`screen` is used only with `.getPrimaryDisplay()` so instead of using `screen`, it could be better to directly create a variable `primaryDisplay`. This way it will reduce the code. You could even further create variables for `workAreaSize` like what you did with `scaleFactor` 👍 

I am personally not very fond of the way `displaySize` and `offset` are set as objects as this feels unnecessary and will allocate more memory. Writing these as variable and ordering them in the code vertically instead of as right now horizontally would make the code a lot more readable. 

Here is an example: 

```
displayWidth = workAreaSize.width,
displayHeight = workAreaSize.height - taskBarHeight,
offsetX = displayWidth - ((trayPosition.x + (trayPosition.width / 2)) / scaleFactor),
offsetY = displayHeight - ((trayPosition.y - (trayPosition.height / 2)) / scaleFactor)
```